src/data_management/rasters/RasterForm.js

```diff
@@ -773,6 +778,64 @@ class RasterFormModel extends Component {
                     />
                   }
                 />
+                <GenericStep
+                  titleComponent={
+                    <FormattedMessage
+                      id="rasters.public_private"
+                      defaultMessage="Who can read"
+                    />
+                  }
+                  subtitleComponent={
+                    <FormattedMessage
+                      id="rasters.public_common_private"
+                      defaultMessage="Specify who can view this raster"
+                    />
+                  }
+                  step={3} // int for denoting which step of the GenericTextInputComponent it refers to
+                  opened={this.props.currentRaster || currentStep === 3}
+                  readonly={false}
+                  currentStep={currentStep} // int for denoting which step is currently active
+                  setCurrentStep={this.setCurrentStep} // cb function for updating which step becomes active
+                  showCheckMark={
+                    this.state.sharedWithOrganisationsWasEverOpenedByUser
+                  }
+                  showNextButton={!this.props.currentRaster}
+                  fields={
+                    <SelectBoxSimple
+                      choices={[
+                        {
+                          name: "Private",
+                          info: "Just the selected organisations can view"
+                        },
+                        {
+                          name: "Common",
+                          info: "All logged in users can view"
+                        },
+                        {
+                          name: "Public",
+                          info: "All users can view"
+                        }
+                      ]}
+                      choice={this.state.accesModifier}
+                      isFetching={false}
+                      transformChoiceToDisplayValue={e => e.name}
+                      updateModelValue={e =>
+                        this.setState({ accesModifier: e.name })}
+                      onKeyUp={e =>
+                        // for now do nothing
+                        // commenting alltogether will crash the app
+                        // TODO implement handle enter
+                        // this.handleEnter(e)
+                        e}
+                      // inputId={titleComponent.props.id + "_input"}
+                      // placeholder={placeholder}
+                      // validate={validate}
+                      transformChoiceToDescription={e => e.info}
+                      // transformChoiceToInfo={e=> e.info}
+                      // noneValue={this.props.noneValue}
```

Comment:
Ps. Er staan nog wat properties uitgecomment.
Is dit om aan te geven wat er mogelijk nog veranderd zou kunnen worden, of als overzicht voor wat het component zou kunnen verwachten?

Reply:
Aangepast